From: Venkateswara Naralasetty <vnaralas@codeaurora.org>
Date: Sun, 14 Jun 2020 09:47:23 +0200
Subject: ath10k : Fix channel survey dump

Channel active/busy time are showing incorrect(less than previous or
sometimes zero) for successive survey dump command.

example:
Survey data from wlan0
	frequency:                      5180 MHz [in use]
	channel active time:            54995 ms
	channel busy time:              432 ms
	channel receive time:           0 ms
	channel transmit time:          59 ms
Survey data from wlan0
	frequency:                      5180 MHz [in use]
	channel active time:            32592 ms
	channel busy time:              254 ms
	channel receive time:           0 ms
	channel transmit time:          0 ms

This patch fix this issue by assigning 'wmi_bss_survey_req_type'
as 'WMI_BSS_SURVEY_REQ_TYPE_READ' which accumulate survey data in
FW and send survey data to driver upon the driver request. Wrap around
is taken care by FW.

hardware used : QCA9984
firmware ver  : ver 10.4-3.5.3-00057

diff --git a/package/kernel/ath10k-ct/patches/9585-ath10k-Fix-channel-survey-dump.patch b/package/kernel/ath10k-ct/patches/9585-ath10k-Fix-channel-survey-dump.patch
new file mode 100644
index 0000000000000000000000000000000000000000..16339a8aba92d2e3748e1b42c8354bcdf3b78a48
--- /dev/null
+++ b/package/kernel/ath10k-ct/patches/9585-ath10k-Fix-channel-survey-dump.patch
@@ -0,0 +1,44 @@
+From: Venkateswara Naralasetty <vnaralas@codeaurora.org>
+Date: Tue, 31 Jul 2018 17:41:17 +0530
+Subject: ath10k : Fix channel survey dump
+
+Channel active/busy time are showing incorrect(less than previous or
+sometimes zero) for successive survey dump command.
+
+example:
+Survey data from wlan0
+	frequency:                      5180 MHz [in use]
+	channel active time:            54995 ms
+	channel busy time:              432 ms
+	channel receive time:           0 ms
+	channel transmit time:          59 ms
+Survey data from wlan0
+	frequency:                      5180 MHz [in use]
+	channel active time:            32592 ms
+	channel busy time:              254 ms
+	channel receive time:           0 ms
+	channel transmit time:          0 ms
+
+This patch fix this issue by assigning 'wmi_bss_survey_req_type'
+as 'WMI_BSS_SURVEY_REQ_TYPE_READ' which accumulate survey data in
+FW and send survey data to driver upon the driver request. Wrap around
+is taken care by FW.
+
+hardware used : QCA9984
+firmware ver  : ver 10.4-3.5.3-00057
+
+Signed-off-by: Venkateswara Naralasetty <vnaralas@codeaurora.org>
+
+Origin: other, https://patchwork.kernel.org/patch/10550707/
+
+--- a/ath10k-4.19/mac.c
++++ b/ath10k-4.19/mac.c
+@@ -7954,7 +7954,7 @@ ath10k_mac_update_bss_chan_survey(struct
+ 				  struct ieee80211_channel *channel)
+ {
+ 	int ret;
+-	enum wmi_bss_survey_req_type type = WMI_BSS_SURVEY_REQ_TYPE_READ_CLEAR;
++	enum wmi_bss_survey_req_type type = WMI_BSS_SURVEY_REQ_TYPE_READ;
+ 
+ 	lockdep_assert_held(&ar->conf_mutex);
+ 
